baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LambdaMethodReference.java

```diff
@@ -91,49 +89,97 @@ public Description matchLambdaExpression(LambdaExpressionTree tree, VisitorState
+                .addFix(buildFix(methodSymbol, methodInvocation, root, state, isLocal))
                 .build();
     }
 
-    private static boolean shouldIgnore(
-            Symbol.MethodSymbol methodSymbol, LambdaExpressionTree root, MethodInvocationTree methodInvocation) {
-        if (!methodSymbol.isStatic()) {
-            if (root.getParameters().size() == 1) {
-                Symbol paramSymbol = ASTHelpers.getSymbol(Iterables.getOnlyElement(root.getParameters()));
-                Symbol receiverSymbol = ASTHelpers.getSymbol(ASTHelpers.getReceiver(methodInvocation));
-                return !paramSymbol.equals(receiverSymbol);
-            }
-            return true;
+    private Description convertMethodInvocations(
+            Symbol.MethodSymbol methodSymbol,
+            MethodInvocationTree methodInvocation,
+            LambdaExpressionTree root,
+            VisitorState state) {
+        List<Symbol> methodParams = getSymbols(methodInvocation.getArguments());
+        List<Symbol> lambdaParam = getSymbols(root.getParameters());
+
+        // We are guaranteed that all of root params are symbols so equality should handle cases where methodInvocation
+        // arguments are not symbols or are out of order
+        if (!methodParams.equals(lambdaParam)) {
+            return Description.NO_MATCH;
         }
-        return !root.getParameters().isEmpty();
+
+        boolean isLocal = ASTHelpers.getReceiver(methodInvocation) == null;
+        return buildDescription(root)
+                .setMessage(MESSAGE)
+                .addFix(buildFix(methodSymbol, methodInvocation, root, state, isLocal))
+                .build();
+    }
+
+    private static List<Symbol> getSymbols(List<? extends Tree> params) {
+        return params.stream()
+                .map(ASTHelpers::getSymbol)
+                .filter(Objects::nonNull)
+                .collect(ImmutableList.toImmutableList());
     }
 
     private static Optional<SuggestedFix> buildFix(
             Symbol.MethodSymbol symbol,
             MethodInvocationTree invocation,
             LambdaExpressionTree root,
-            VisitorState state) {
+            VisitorState state,
+            boolean isLocal) {
         SuggestedFix.Builder builder = SuggestedFix.builder();
-        return toMethodReference(qualifyTarget(symbol, invocation, builder, state))
+        return toMethodReference(qualifyTarget(symbol, invocation, builder, state, isLocal))
                 .map(qualified -> builder.replace(root, qualified).build());
     }
 
     private static String qualifyTarget(
             Symbol.MethodSymbol symbol,
             MethodInvocationTree invocation,
             SuggestedFix.Builder builder,
-            VisitorState state) {
+            VisitorState state,
+            boolean isLocal) {
+        if (!symbol.isStatic() && isLocal) {
+            return "this." + symbol.name.toString();
```

Comment:
I think there might be some oddness if the method is. It provided by this, but an enclosing class, requiring prefix “Enclosing.this.“. Worth a test to verify, I’d probably exclude that from the check rather than attempt to refactor since it’s fairly uncommon.